examples/development/compose/init.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require "portus/db"
+
+#
+# First of all, wait until the database is up and running. This is useful in
+# containerized scenarios.
+#
+
+count = 0
+TIMEOUT = 90
+
+while (status = ::Portus::DB.ping) != :ready
+  if count >= TIMEOUT
+    Rails.logger.tagged("Database") do
+      Rails.logger.error "Timeout reached, exiting with error. Check the logs..."
+    end
+    exit 1
+  end
+
+  Rails.logger.tagged("Database") { Rails.logger.error "Not ready yet. Waiting..." }
+  sleep 5
+  count += 5
+
+  system("bundle exec rake db:setup") if status == :missing
+end
```

Comment:
You might want to put this code in `lib/portus/db` as a method, and then use this both here and in the background.rb file 😉

Reply:
And be able to pass a block to it? That sounds interesting...